test/serverTest.js

```diff
@@ -50,3 +53,153 @@ describe('server', () => {
+    let viewsBefore = 0;
+    let viewsAfter = 0;
+    postgresDb.movieDb.query(getViews)
+      .then((resultBefore) => {
+        viewsBefore = resultBefore[0][0].views;
+        chai.request('http://localhost:3000')
+          .post('/tetraflix/sessionData')
+          .set('content-type', 'application/json')
+          .send({
+            userId: 534356757834,
+            events: [{
+              movie: {
+                id: movieId,
+              },
+              progress: 1,
+              timestamp: '2017-09-08 12:50PM',
+            }],
+          })
+          .end((err, res) => {
+            if (err) {
+              done(err);
+            }
+            postgresDb.movieDb.query(getViews)
+              .then((resultAfter) => {
+                viewsAfter = resultAfter[0][0].views;
+                viewsAfter.should.equal(viewsBefore + 1);
+              });
+          });
+        done();
+      });
+  });
+
+  it('should not update view count when progress is less than 1', (done) => {
+    const movieId = Math.floor(Math.random() * 300000);
+    const getViews = `select views from movies where id = ${movieId}`;
+    let viewsBefore = 0;
+    let viewsAfter = 0;
+    postgresDb.movieDb.query(getViews)
+      .then((resultBefore) => {
+        viewsBefore = resultBefore[0][0].views;
+        chai.request('http://localhost:3000')
+          .post('/tetraflix/sessionData')
+          .set('content-type', 'application/json')
+          .send({
+            userId: 12345,
+            events: [{
+              movie: {
+                id: movieId,
+              },
+              progress: 0.9,
+              timestamp: '2017-09-08 12:50PM',
+            }],
+          })
+          .end((err, res) => {
+            if (err) {
+              done(err);
+            }
+            postgresDb.movieDb.query(getViews)
+              .then((resultAfter) => {
+                viewsAfter = resultAfter[0][0].views;
+                viewsAfter.should.equal(viewsBefore);
+              });
+          });
+        done();
+      });
+  });
+
+  it('should update view count for multiple movies', (done) => {
+    const rand = Math.floor(Math.random() * 100000);
+    const movie1 = rand;
+    const movie2 = rand + 100000;
+    const movie3 = rand + 200000;
+    const sendObj = {
+      userID: 12345,
+      events: [
+        {
+          movie: {
+            id: movie1,
+          },
+          progress: 1,
+          timestamp: '2017-09-08 12:50PM',
+        },
+        {
+          movie: {
+            id: movie2,
+          },
+          progress: 1,
+          timestamp: '2017-09-08 12:50PM',
+        },
+        {
+          movie: {
+            id: movie3,
+          },
+          progress: 0.9,
+          timestamp: '2017-09-08 12:50PM',
+        },
+      ],
+    };
+    let movie1Before = 0;
+    let movie2Before = 0;
+    let movie3Before = 0;
+    let movie1After = 0;
```

Comment:
I think it maybe clearer to not initialize movie1After/movie2After/movie3After to 0 here since the these values of these variables will be based on the response you get in lines 194-196.

Reply:
done.  I also changed this in the previous tests as well.